src/bin/stratisd.rs

```diff
@@ -79,54 +116,103 @@ fn run() -> StratisResult<()> {
         }
     };
 
-    #[cfg(feature="dbus_enabled")]
-    let (dbus_conn, mut tree, dbus_context) = libstratis::dbus_api::connect(Rc::clone(&engine))?;
+    /*
+    The file descriptor array indexes are laid out in the following:
+
+    0   == Always udev fd index
+    1   == engine index if eventable
+    1/2 == Start of dbus client file descriptor(s), 1 if engine is not eventable, else 2
+    */
+    const FD_INDEX_UDEV: usize = 0;
+    const FD_INDEX_ENGINE: usize = 1;
+
 
     let mut fds = Vec::new();
 
+    fds.push(libc::pollfd {
+                 fd: udev.as_raw_fd(),
+                 revents: 0,
+                 events: libc::POLLIN,
+             });
+
     let mut eventable = engine.borrow_mut().get_eventable()?;
-    let engine_fds_end_idx = match eventable {
+
+    // The variable _dbus_client_index_start is only used when dbus support is compiled in, thus
+    // we denote the value as not needed to compile when dbus support is not included.
+    let (engine_eventable, poll_timeout, _dbus_client_index_start) = match eventable {
         Some(ref mut evt) => {
             fds.push(libc::pollfd {
                          fd: evt.get_pollable_fd(),
                          revents: 0,
                          events: libc::POLLIN,
                      });
-            1
+
+            // Don't timeout if eventable, we are event driven
+            (true, -1, FD_INDEX_ENGINE + 1)
         }
-        None => 0,
+
+        // We periodically need to timeout as we are not event driven
+        None => (false, 10000, FD_INDEX_ENGINE),
     };
 
-    // Don't timeout if eventable lets us be event-driven
-    let poll_timeout = if engine_fds_end_idx != 0 { -1 } else { 10000 };
+    #[cfg(feature="dbus_enabled")]
+    let (mut dbus_conn, mut tree, base_object_path, mut dbus_context) =
+        libstratis::dbus_api::connect(Rc::clone(&engine))?;
 
     loop {
-        // Unconditionally call evented() if engine has no eventable.
-        // This looks like a bad idea, but the only engine that has
-        // no eventable is the sim engine, and for that engine, evented()
-        // is essentially a no-op.
-        if engine_fds_end_idx == 0 {
-            engine.borrow_mut().evented()?;
+        // Process any udev block events
+        if fds[FD_INDEX_UDEV].revents != 0 {
+            loop {
+                match udev.receive_event() {
+                    Some(event) => {
+                        if let Some((devnode, device)) = handle_udev_add(&event) {
+                            let pool_uuid = engine.borrow_mut().block_evaluate(devnode, device);
```

Comment:
To ignore the possible error:

```
engine.borrow_mut().block_evaluate(devnode, device).unwrap_or(None);
```

and maybe an explanation of why it is ok to ignore it. That explanation might be that, if it is not our device, obviously there is nothing to do. If it our device yet we missed it, continuing with execution is still better than failing, as the worst that can happen is that we'ld be stuck forever with an incomplete pool.